specs/WebAuthenticationRequested.md

```diff
@@ -0,0 +1,270 @@
+                args->GetDeferral(&deferral);
+                ShowCustomLoginUI().then([web_auth_args, deferral](LPCWSTR userName, LPCWSTR password) {
+                    wil::com_ptr<ICoreWebView2BasicAuthenticationResponse> basicAuthenticationResponse;
+                    args->get_Response(&basicAuthenticationResponse);
+                    basicAuthenticationResponse->put_UserName(userName);
+                    basicAuthenticationResponse->put_Password(password);
+                    deferral->Complete();
+                }
+                
+                return S_OK;
+            })
+            .Get(),
+        &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    Credential credential = await ShowCustomLoginUIAsync();
+    args.Response.UserName = credential.UserName;
+    args.Response.Password = credential.Password;
+    deferral.Complete();
+};
+```
+
+## Cancel authentication prompt
+The developer can block the authentication request. In this case, the default login dialog prompt will no longer be shown to the user and the server will respond as if the user clicked cancel.
+
+```cpp
+    webview2->add_BasicAuthenticationRequested(
+        Callback<ICoreWebView2BasicAuthenticationRequestedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2BasicAuthenticationRequestedEventArgs* args)
+            {
+                args->put_Cancel(true);
+
+                return S_OK;
+            })
+            .Get(),
+        &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    args.Cancel = true;
+};
+```
+
+## Read authentication challenge string
+Developer can read the authentication challenge string sent by server. Note that if the developer doesn't cancel or provide a response, the default login dialog prompt will be shown to the user.
+
+```cpp
+webview2->add_BasicAuthenticationRequested(
+    Callback<ICoreWebView2BasicAuthenticationRequestedEventHandler>(
+        [this](
+            ICoreWebView2* sender,
+            ICoreWebView2BasicAuthenticationRequestedEventArgs* args)
+        {
+            args->get_Challenge(&challenge);
+            if (!ValidateChallenge(challenge.get())) { // Check the challenge string
+                args->put_Cancel(true);
+            }
+            return S_OK;
+        })
+        .Get(),
+    &m_BasicAuthenticationRequestedToken));
+```
+
+```c#
+webView.CoreWebView2.BasicAuthenticationRequested += delegate (object sender, CoreWebView2BasicAuthenticationRequestedEventArgs args)
+{
+    if (args.Challenge.Equals("Expected login credentials")) {
```

Comment:
Is 'Expected login credentials' a real sort of challenge value? A separate method like ValidateChallenge like you have in the previous example may be better than showing a fake challenge value